### PR TITLE
Adding Azure Data lake export to Streaming pipelines (Delta Table)

### PR DESCRIPTION
**Pull request by Luishfs**
# Summary
Adding Azure Data lake export to Streaming pipelines (Delta Table)
Added docs

# Tests
Tested using RabbitMQ source to ADLS, in both local and Azure Storage
Mage logs:
Screenshot from 2023-07-12 21-04-00

Azure Storage:
Screenshot from 2023-07-12 21-03-19

cc:
@wangxiaoyou1993

### CODE REVIEW

File: mage_ai/data_preparation/templates/data_exporters/streaming/azure_data_lake.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+connector_type: azure_data_lake
+table_uri: abfs://container@storagename.dfs.core.windows.net/
+account_name: account_name
+access_key: access_key
+mode: append
+file_type: delta
```
Comment (wangxiaoyou1993) [inline mage_ai/data_preparation/templates/data_exporters/streaming/azure_data_lake.yaml:6]: nit: add newline to end of the file

File: mage_ai/streaming/sinks/azure_data_lake.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        self._print(f'Upload {len(self.buffer)} records to DeltaTable.')
+
+        df = pd.DataFrame(self.buffer)
+
+        if self.config.file_type == 'delta':
+            curr_time = datetime.now(timezone.utc)
+            filename = curr_time.strftime('%Y%m%d-%H%M%S')
+            try:
+                storage_options = {
+                    "azure_storage_account_name": self.config.account_name,
```
Comment (wangxiaoyou1993) [inline mage_ai/streaming/sinks/azure_data_lake.py:52]: We prefer using single quotes in the code.

File: mage_ai/frontend/components/PipelineDetail/AddNewBlocks/utils.tsx [hunk 3]
```diff
@@ -44,6 +44,7 @@ const getDataSourceTypes = (
         DataSourceTypeEnum.KINESIS,
         DataSourceTypeEnum.MONGODB,
         DataSourceTypeEnum.OPENSEARCH,
+        DataSourceTypeEnum.AZURE_DATA_LAKE,
```
Comment (wangxiaoyou1993) [inline mage_ai/frontend/components/PipelineDetail/AddNewBlocks/utils.tsx:47]: Let's sort them in alphabetical order

File: mage_ai/data_preparation/templates/data_exporters/streaming/azure_data_lake.yaml (hunk 1 above)
Comment (Luishfs) [reply mage_ai/data_preparation/templates/data_exporters/streaming/azure_data_lake.yaml]: fixed on 23cc8513f148e9eb7e07fd3d283fe472ea62eae5

File: mage_ai/streaming/sinks/azure_data_lake.py (hunk 2 above)
Comment (Luishfs) [reply mage_ai/streaming/sinks/azure_data_lake.py]: fixed on 23cc8513f148e9eb7e07fd3d283fe472ea62eae5

File: mage_ai/frontend/components/PipelineDetail/AddNewBlocks/utils.tsx (hunk 3 above)
Comment (Luishfs) [reply mage_ai/frontend/components/PipelineDetail/AddNewBlocks/utils.tsx]: sorted on 23cc8513f148e9eb7e07fd3d283fe472ea62eae5

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**